py/desispec/scripts/proc_joint_fit.py

```diff
@@ -230,84 +230,97 @@ def main(args=None, comm=None):
-                                pg = "CALIB DESI-CALIB-0{} LEDs only".format(lampbox)
-
-                                cmd = "desi_average_fiberflat --program '{}' --outfile {} -i ".format(pg, ofile)
-                                for flat in flats_for_this_camera:
-                                    cmd += " {} ".format(flat)
-                                num_cmd += 1
-                                err = runcmd(cmd, inputs=flats_for_this_camera, outputs=[ofile, ])
-                                if err:
-                                    num_err += 1
-                                if os.path.isfile(ofile):
-                                    average_flats[camera].append(ofile)
-                                else:
-                                    log.error(f"Generating {ofile} failed; proceeding with other flats")
-                            else:
-                                log.info("Will use existing {}".format(ofile))
-                                average_flats[camera].append(ofile)
-
-                    log.info("Auto-calibration across lamps and spectro  per camera arm (b,r,z)")
-                    for camera_arm in ["b", "r", "z"]:
-                        cameras_for_this_arm = []
-                        flats_for_this_arm = []
-                        for camera in args.cameras:
-                            if camera[0].lower() == camera_arm:
-                                cameras_for_this_arm.append(camera)
-                                if camera in average_flats:
-                                    for flat in average_flats[camera]:
-                                        flats_for_this_arm.append(flat)
-                        if len(flats_for_this_arm) > 0:
-                            cmd = "desi_autocalib_fiberflat --night {} --arm {} -i ".format(args.night, camera_arm)
-                            for flat in flats_for_this_arm:
-                                cmd += " {} ".format(flat)
-                            num_cmd += 1
-                            err = runcmd(cmd, inputs=flats_for_this_arm, outputs=[])
-                            if err:
-                                num_err += 1
-                        else:
-                            log.error(f'No flats found for arm {camera_arm}')
-
-                    log.info("Done with fiber flats per night")
+                log.info(f"Rank {rank} will use existing {ofile}")
+
+        if comm is not None:
+            comm.barrier()
+
+        log.info("Auto-calibration across lamps and spectro  per camera arm (b,r,z)")
+        for camera_arm in ["b", "r", "z"][rank::size]:
+            log.info(f"Rank {rank} autocalibrating across spectro for camera arm {camera_arm}")
+            cmd = f"desi_autocalib_fiberflat --night {args.night} --arm {camera_arm} -i "
+            for flat in flats_for_arm[camera_arm]:
+                cmd += f" {flat} "
```

Comment:
Could remove the trailing space here and in line 314, though it shouldn't matter

Reply:
It shouldn't matter for the final command, and it was semi-unintentional (from me, I was copying earlier code from Daniel who may have gotten it from you :) ), but it provides the epsilon of safety that if another line of code appends another option it still works even if they forget to include a leading space.  I'll use that to justify leaving it as is :)

FWIW, I had also considered
```
cmd += ' '.join(flats_for_arm[camera_arm])
```
but that seems more clever than obvious.